### PR TITLE
MTDSA-3732 Added connector and parser for create loss claims

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: app/v1/connectors/LossClaimsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@Singleton
+class LossClaimsConnector @Inject()(http: HttpClient, appConfig: AppConfig) extends DesConnector {
+
+  def createLossClaim(createLossClaimsRequest: CreateLossClaimsRequest)(implicit hc: HeaderCarrier,
+                                                                         ec: ExecutionContext): Future[DesOutcome[CreateLossClaimsResponse]] = {
+    val nino = createLossClaimsRequest.nino.nino
+
+    def doIt(implicit hc: HeaderCarrier) =
+      http.POST[LossClaim, DesOutcome[CreateLossClaimsResponse]](s"${appConfig.desBaseUrl}/income-tax/$nino/loss-claims",
```
Comment (jeremystone) [inline app/v1/connectors/LossClaimsConnector.scala]: URI should be `/income-tax/claims-for-relief/{taxableEntityId}`

File: app/v1/models/requestData/CreateLossClaimsRequest.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.requestData
+
+import uk.gov.hmrc.domain.Nino
+import v1.models.domain.LossClaim
+
+case class CreateLossClaimsRequest(nino: Nino, lossClaim: LossClaim)
```
Comment (jeremystone) [inline app/v1/models/requestData/CreateLossClaimsRequest.scala]: Singular CreateLossClaimRequest (to match CreateBFLossRequest)

File: app/v1/models/des/CreateLossClaimsResponse.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.des
+
+import play.api.libs.json.{Json, Reads, Writes, __}
+
+case class CreateLossClaimsResponse(id: String)
```
Comment (jeremystone) [inline app/v1/models/des/CreateLossClaimsResponse.scala]: Singular: CreateLossClaimResponse

File: app/v1/connectors/BFLossConnector.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import v1.connectors.httpparsers.StandardDesHttpParser._
+import v1.models.des.{AmendBFLossResponse, CreateBFLossResponse, ListBFLossesResponse, RetrieveBFLossResponse}
+import v1.models.domain.{AmendBFLoss, BFLoss}
+import v1.models.requestData._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class BFLossConnector @Inject()(http: HttpClient, appConfig: AppConfig) extends DesConnector {
```
Comment (jeremystone) [inline app/v1/connectors/BFLossConnector.scala]: (Minor) Either BFLossesConnector to match LossClaimsConnector, or rename LossClaimsConnector to LossClaimConnector for consistency.

Comment (narendravijayarao) [reply app/v1/connectors/BFLossConnector.scala]: why? they both are completely different... I created based on the URL in tech spec

File: app/v1/models/des/CreateLossClaimsResponse.scala (hunk 3 above)
Comment (narendravijayarao) [reply app/v1/models/des/CreateLossClaimsResponse.scala]: A loss can have multiple claims, same as per the URL

File: app/v1/models/requestData/CreateLossClaimsRequest.scala (hunk 2 above)
Comment (narendravijayarao) [reply app/v1/models/requestData/CreateLossClaimsRequest.scala]: same as below

File: app/v1/models/des/CreateLossClaimsResponse.scala (hunk 3 above)
Comment (jeremystone) [reply app/v1/models/des/CreateLossClaimsResponse.scala]: Yes. But I understood this request to be for a single claim.

**Pull request merged**